File: excel-api-java/src/main/java/pl/alyx/api/excel/util/DurationParser.java
package pl.alyx.api.excel.util;

import java.time.Duration;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class DurationParser {
    private static final Pattern DURATION_PATTERN = Pattern.compile("^(\\d+)([smh])$");

    public static Duration parse(String duration) {
        Matcher matcher = DURATION_PATTERN.matcher(duration);
        if (!matcher.matches()) {
            throw new IllegalArgumentException(
                "Invalid duration format: " + duration + ". Expected format: <number><unit> where unit is s, m, or h."
            );
        }
        long value = Long.parseLong(matcher.group(1));
        String unit = matcher.group(2);
        return switch (unit) {
            case "s" -> Duration.ofSeconds(value);
            case "m" -> Duration.ofMinutes(value);
            case "h" -> Duration.ofHours(value);
            default -> throw new IllegalArgumentException(
                "Invalid duration unit: " + unit + ". Expected s, m, or h."
            );
        };
    }
}

File: excel-api-csharp/src/ExcelApi/Util/DurationParser.cs
using System;
using System.Text.RegularExpressions;

namespace BigBytes.ExcelApi.Util;

public static class DurationParser
{
    private static readonly Regex DurationPattern = new Regex(@"^(\d+)([smh])$", RegexOptions.Compiled);

    public static TimeSpan Parse(string duration)
    {
        var match = DurationPattern.Match(duration);
        if (!match.Success)
        {
            throw new ArgumentException($"Invalid duration format: {duration}. Expected format: <number><unit> where unit is s, m, or h.");
        }

        long value = long.Parse(match.Groups[1].Value);
        string unit = match.Groups[2].Value;

        return unit switch
        {
            "s" => TimeSpan.FromSeconds(value),
            "m" => TimeSpan.FromMinutes(value),
            "h" => TimeSpan.FromHours(value),
            _ => throw new ArgumentException($"Invalid duration unit: {unit}. Expected s, m, or h.")
        };
    }
}
